packages/ldap/test/helpers.ts

```diff
@@ -1,18 +1,19 @@
-import type { Task, Resource } from 'effection';
+import type { Task, Stream, Operation } from 'effection';
+import { createStream, ensure, on, once, spawn } from 'effection';
 import type { Client } from '@simulacrum/client';
 import { createClient } from '@simulacrum/client';
 import type { Server, ServerOptions } from '@simulacrum/server';
 import { createSimulationServer } from '@simulacrum/server';
-
 export type { Client, Simulation } from '@simulacrum/client';
+import { createClient as createLDAPJSClient, SearchEntryObject, SearchOptions } from 'ldapjs';
 
-export function createTestServer(options: ServerOptions): Resource<Client> {
+export function createTestServer(options: ServerOptions): Operation<Client> {
   return {
     *init(scope: Task) {
       let server: Server = yield createSimulationServer(options);
       let { port } = server.address;
       let client = createClient(`http://localhost:${port}`);
-      scope.run(function*() {
+      yield scope.spawn(function*() {
```

Comment:
We should always use `scope.spawn()` since it is an operation rather than a synchronous method.

Reply:
What about an eslint for things like this?

Reply:
eslint could keep you developers like me on the straight and narrow in effection

Reply:
@dagda1 That would be amazing actually. Another option is to straight up deprecate `Task#run()` in favor of having a some sort of `runWithin(operation, task)` In case you need to spawn a task synchronously from outside of an operation.